### PR TITLE
Create Hash-table.md

### PR DESCRIPTION
**Pull request by AmrutaJayanti**
## Related Issue

Fix: #318 

## Description

Created `Hash-table.md`

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [ ] Documentation update
- [x] Other (specify): ______Add Documentation______

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: dsa/intermediate/Hash-table.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Check the corresponding bucket. If using chaining, search the linked list. If using open addressing, follow the probing sequence until the key is found or an empty bucket is reached.
+
+**Deletion:**
+
+Compute the hash code of the key.
+Map the hash code to an index.
+Locate the key in the bucket and remove the key-value pair. Adjust the structure to maintain efficiency (e.g., rehash elements if necessary).
+
+## Complexity
+Average Time Complexity: O(1) for insert, delete, and search operations in a well-implemented hash table with a good hash function and a load factor that avoids excessive collisions.
```
Comment (Ajay-Dhangar) [inline dsa/intermediate/Hash-table.md:35]: replace `O(1)` to `$O(1)$`

File: dsa/intermediate/Hash-table.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Worst-Case Time Complexity: O(n) for insert, delete, and search operations if all keys hash to the same index (highly unlikely with a good hash function).
```
Comment (Ajay-Dhangar) [inline dsa/intermediate/Hash-table.md:37]: replace `O(n)` to `$O(n)$`

File: dsa/intermediate/Hash-table.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+## Load Factor
+The load factor (α) is the ratio of the number of entries to the number of buckets in the hash table. A high load factor can lead to more collisions, affecting performance. A common practice is to resize the hash table (rehash) when the load factor exceeds a certain threshold.
```
Comment (Ajay-Dhangar) [inline dsa/intermediate/Hash-table.md:40]: replace `(α)` to `$(α)$`

File: dsa/intermediate/Hash-table.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+
+​Load Factor = No. of entries /  No. of Buckets
```
Comment (Ajay-Dhangar) [inline dsa/intermediate/Hash-table.md:43]: Replace `​Load Factor = No. of entries / No. of Buckets` to `$$​Load Factor = No. of entries / No. of Buckets$$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Now change some code based on suggestions

**Comment by AmrutaJayanti** [comment]
@Ajay-Dhangar I made the changes .Please approve

**Review by Ajay-Dhangar — APPROVED** [review]
okay

**Pull request merged**